### PR TITLE
Fix some issues with the import, specifially the SPDX SBOM import

### LINKED ISSUE #498: Running the redhat-sbom importer generates an error while importing data
Steps to reproduce:
- Start a fully clean instance of Trustify
- Enable the provided importer named `redhat-sbom`. Let it run for some minutes and the importer should generate a report `GET api/v1/importer/redhat-sbom/report` with the following response:

```json
{
    "items": [
        {
            "id": "e80ec858-8fa8-4f88-9ad5-e786d461efe5",
            "importer": "redhat-sbom",
            "creation": "2024-07-02T13:22:24.108607Z",
            "error": "Visitor error: Execution Error: error returned from database: insert or update on table \"package_relates_to_package\" violates foreign key constraint \"package_relates_to_package_sbom_id_left_node_id_fkey\"",
            "report": {
                "endDate": "2024-07-02T13:22:24.105880274Z",
                "messages": {
                    "Upload": {
                        "https://access.redhat.com/security/data/sbom/beta/spdx/rhel-8.8.0.json.bz2": [
                            {
                                "message": "upload failed: Execution Error: error returned from database: insert or update on table \"package_relates_to_package\" violates foreign key constraint \"package_relates_to_package_sbom_id_left_node_id_fkey\"",
                                "severity": "Error"
                            }
                        ]
                    }
                },
                "startDate": "2024-07-02T13:21:15.871547194Z",
                "numerOfItems": 2
            }
        }
    ],
    "total": 1
}
```

You can see there is an error `insert or update on table \"package_relates_to_package\" violates foreign key constraint \"package_relates_to_package_sbom_id_left_node_id_fkey\"`

### PR DESCRIPTION
**Pull request by ctron**
The catalyst of this was https://github.com/trustification/trustify/issues/498 … working on this uncovered a few extra issues.

The main motivation was to fix the issue that an SBOM with broken references would (rightfully) cause a DB foreign key error. This issue can be be detected and a proper error message can be provided.

However, it also turned out that such an error would prevent the import from continuing and would simply abort. This is also fixed now, and such errors will be recorded as part of the report and stored in the DB at the end up the run.

Furthermore, the jobs used the "last run" time for the next run (as "since"). Which leads to the case that the previous documents, skipped due to the error, would be skipped again. This is changed in way that only a successful run can provide a timestamp for the next run. (Critically) failed runs will cause a complete re-run the next time the job is run.

Closes #498

### CODE REVIEW

**Comment by ctron** [comment]
@carlosthe19916 this is the PR with requires https://github.com/trustification/trustify-ui/pull/101

**Comment by ctron** [comment]
Blocked by https://github.com/KenDJohnson/cpe-rs/pull/15

**Comment by ctron** [comment]
> Blocked by [KenDJohnson/cpe-rs#15](https://github.com/KenDJohnson/cpe-rs/pull/15)

Created a PR and set that as a patch

**Comment by ctron** [comment]
rebased

File: etc/test-data/broken-refs.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+  "SPDXID": "SPDXRef-DOCUMENT",
```
Comment (bobmcwhirter) [inline etc/test-data/broken-refs.json:2]: nit: can we stuff this into `test-data/spdx/...`

File: modules/ingestor/src/graph/cpe.rs [hunk 2]
```diff
@@ -117,3 +124,57 @@ impl Debug for CpeContext {
...
     }
 }
+
+#[derive(Debug)]
+pub struct CpeCreator {
+    graph: Graph,
+    cpes: HashMap<Cpe, CpeContext>,
+}
+
+impl CpeCreator {
```
Comment (bobmcwhirter) [inline modules/ingestor/src/graph/cpe.rs:134]: Nice

**Review by bobmcwhirter — APPROVED** [review]

**Pull request merged**